animal_crossing.py

```diff
@@ -0,0 +1,187 @@
+import dateutil.tz
+
+import config
+
+if config.bot.acnh_db is not None:
+	db = sqlite3.connect(config.bot.acnh_db)
+	db.row_factory = sqlite3.Row
+
+	# enable foreign key constraints
+	with db:
+		db.execute('PRAGMA foreign_keys = ON')
+
+time_format = '%Y-%m-%d %H:%M'
+cached_row_id = 0
+
+def stalk_market(cmd):
+	if not cmd.args:
+		cmd.reply('''Please register your friend code with '!fc set <friend-code>' before using the following commands:
+- !stalks tz <tz>: will set your local timezone. See https://en.wikipedia.org/wiki/List_of_tz_database_time_zones. **This is required for the following commands.**
+- !stalks sell: will list all currently availably offers.
+- !stalks sell <value>: will add a new offer listed at <value> bells.
+- !stalks trigger <value>: will ping you if a new offer is listed above <value> bells. 
+		''')
+		return
+
+	split = cmd.args.split(' ', 1)
+	subcmd = split[0]
+	subargs = len(split) == 2 and split[1] or ''
+	if subcmd == 'tz':
+		_stalk_set_timezone(cmd, subargs)
+	elif subcmd == 'sell':
+		_stalk_set_sell_price(cmd, subargs)
+	elif subcmd == 'trigger':
+		_stalks_set_sell_trigger(cmd, subargs)
+	else:
+		cmd.reply('Unrecognized stalks subcommand %s' % subcmd)
+
+def _stalk_set_sell_price(cmd, price):
+	if not price:
+		_stalk_list_sale_prices(cmd)
+		return
+
+	user_id = cmd.sender['id']
+	current_time = datetime.now(timezone.utc)
+	cur = db.execute('SELECT timezone FROM user WHERE id=?', (user_id,))
+	res = cur.fetchone()
+	if not res:
+		cmd.reply('Could not add sale price. Have you registered a friend code?')
+		return
+	elif res['timezone'] is None:
+		cmd.reply('Could not add sale price. Please register a time zone with !stalks tz')
+		return
+
+	user_time = current_time.astimezone(dateutil.tz.gettz(res['timezone']))
+	if (user_time.hour < 8 or user_time.hour >= 22):
+		cmd.reply('Your shops are closed. Your current time zone is %s, where it is currently %s.' %
+			(res['timezone'], user_time.strftime(time_format)))
+		return
+	elif user_time.weekday() == 6:
+		cmd.reply('It is currently Sunday in your selected time zone, %s. Turnip offers cannot be submitted.' %
+			(res['timezone']))
+		return
+
+	if user_time.hour >= 12:
+		expiration = user_time.replace(hour=22, minute=0, second=0, microsecond=0)
+	else:
+		expiration = user_time.replace(hour=12, minute=0, second=0, microsecond=0)
+
+	expires_in = _calculate_expiration(expiration - user_time)
+
+	try:
+		value = int(price)
+	except ValueError:
+		cmd.reply('Could not parse sell value. Usage: !stalks sell 123')
+		return
+
+	try:
+		with db:
+			db.execute('INSERT INTO sell_price VALUES (?, ?, ?, ?)',
+				(user_id, current_time, expiration.astimezone(timezone.utc), value))
+		cmd.reply('Sale price recorded at %s bells. Offer expires in %dh %dm.' %
+			(price, expires_in[0], expires_in[1]))
+		_stallk_check_sell_triggers(cmd, price, expires_in)
+	except sqlite3.IntegrityError:
+		cmd.reply('Could not add sale price. Have you registered a friend code?')
```

Comment:
I don't think this is why we'd get an integrity error now since you checked this cond earlier